src/components/Field.tsx

```diff
@@ -0,0 +1,95 @@
+import { 
+	SyntheticEvent,
+	useState,
+	createContext, 
+	useContext,
+	HTMLAttributes,
+} from "react";
+import { pipe } from "fp-ts/lib/function";
+import * as E from "fp-ts/lib/Either";
+import * as A from "fp-ts/lib/Array";
+
+import styles from 'styles/field.module.scss';
+import { ErrorsList } from "helpers/validators";
+import { logPipe } from "helpers/functional";
+
+type InputValue = string | boolean | null;
+type ValidatorProps = { errors: E.Either<string[], string> };
+export type ValidatorType = { [key: string]: () => E.Either<string, string> }[];
+
+type Props = {
+	fieldId: string,
+	label: string,
+	isRequired?: boolean,
+} & HTMLAttributes<HTMLInputElement>
+
+export const FormContext = createContext<ValidatorType>([]);
+
+export default function Field({ fieldId, label, isRequired = false, ...props }: Props) {
+	const validators = useContext<any>(FormContext);
+	const [hasInteracted, setHasInteracted] = useState<boolean>(false);
+	const [data, setData] = useState<InputValue>(null);
+
+	const shouldValidateUser = isRequired && !!validators[fieldId];
+	const validationErrors: E.Either<string[], string> = validators[fieldId]?.(data);
+	const shouldShowFeedback = shouldValidateUser && hasInteracted;
+
+	const setNewValue = (e: SyntheticEvent<HTMLInputElement>) => {
+		const isCheckbox = e.currentTarget.getAttribute('type') === 'checkbox';
+		return setData(isCheckbox ? e.currentTarget.checked : e.currentTarget.value);
+	}
+
+	const formatErrorString = (acc: string, value: ErrorsList[0]) => acc += ` ${value.message}; `;
+	
+	const validateOnBlur = (e: SyntheticEvent<HTMLInputElement>) => {
+		if (!hasInteracted) setHasInteracted(true);
+		setNewValue(e);
+	}
+	
+	const ErrorStringAnnouncer = ({ errors }: { errors: E.Either<ErrorsList, string> }) =>
+		pipe(errors,
+			E.fold(
+				A.reduce('', formatErrorString),
+				() => '',
+			),
+			(str) => {
+				return (
+					<span className={styles.warnText}>{str}</span>
```

Comment:
you can use jsx syntax in fold function

Reply:
Can you give me some example, please? I think in this case I want the same JSX output regardless if it´s Left | Right, so the direct function worked well.

Reply:
ok if JSX the same. E.fold example:
```
  const val: E.Either<Error, string> = Math.random() > 0.5 ? E.left(new Error('left')) : E.right('right');

  return (
    <div>
      {
        pipe(
          val,
          E.fold(
            (e) => <p>{e.message}</p>,
            (s) => <p>{s}</p>
          )
        )
      }
    </div>
```